### PR TITLE
MEPTS-751 - IMER Report: New Indicator IMER1B

### PR DESCRIPTION
**Pull request by iteliolucas**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            cohort,
+            parameterValues,
+            context);
+    CalculationResultMap preARTCareEnrollmentMap = getARTCareEnrollment(cohort, context);
+
+    Date endDate = (Date) parameterValues.get(ON_OR_BEFORE);
+
+    if (endDate == null) {
+      endDate = (Date) context.getFromCache(ON_OR_BEFORE);
+    }
```
Comment (cfaife) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.java]: please use only from getFromCache to retrieve data from UI

Comment (cfaife) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.java]: you need to pass the location as well,

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IMER1BNumeratorCohortQueries.java [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    CalculationCohortDefinition cd =
+        new CalculationCohortDefinition(
+            Context.getRegisteredComponents(
+                    StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.class)
+                .get(0));
+    cd.setName("Initiated ART within 15 days from the ART Care Enrollment");
+    cd.addCalculationParameter("considerTransferredIn", considerTransferredIn);
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+    cd.addParameter(new Parameter("onOrBefore", "Before Date", Date.class));
+    cd.addParameter(new Parameter("onOrAfter", "After Date", Date.class));
```
Comment (cfaife) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IMER1BNumeratorCohortQueries.java]: I believe this is not needed 'onOrAfter'

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+                        new Interval(preARTCareEnrollmentDate.getTime(), artStartDate.getTime()))
+                    .getDays();
+
+            if (days <= INTERVAL_BETWEEN_ART_START_DATE_MINUS_PATIENT_ART_ENROLLMENT_DATE) {
+              lessOrEqualTo15Days = true;
+            }
+          }
+        }
+
+        map.put(patientId, new BooleanResult(lessOrEqualTo15Days, this));
```
Comment (cfaife) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.java]: please change BooleanResult to SimpleResult  to allow us to pass the interval between care start date and art start date

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+        SimpleResult preARTCareEnrollmentResult =
+            (SimpleResult) preARTCareEnrollmentMap.get(patientId);
+
+        if (preARTCareEnrollmentResult != null) {
+          Date preARTCareEnrollmentDate = (Date) preARTCareEnrollmentResult.getValue();
+
+          if (artStartDate != null
+              && preARTCareEnrollmentDate != null
+              && preARTCareEnrollmentDate.getTime() < artStartDate.getTime()) {
```
Comment (cfaife) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.java]: please  preARTCareEnrollmentDate.compareTo(artStartDate)<0

**Review by cfaife — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.java (hunk 4 above)
Comment (iteliolucas) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.java]: Done

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.java (hunk 3 above)
Comment (iteliolucas) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.java]: Done

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IMER1BNumeratorCohortQueries.java (hunk 2 above)
Comment (iteliolucas) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IMER1BNumeratorCohortQueries.java]: Done removing 'onOrAfter''

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.java (hunk 1 above)
Comment (iteliolucas) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.java]: Done

**Review by cfaife — APPROVED** [review]

**Pull request merged**